lib/Doctrine/entities/User.php

```diff
@@ -279,9 +279,17 @@ public function setUsername1($username1) {
     /**
      * Set the user's home Site, Legacy and optional.
      * @deprecated since version 5.4
-     * @param Site $homeSite
+     * @param mixed $homeSite
      */
-    public function setHomeSiteDoJoin(Site $homeSite) {
+    public function setHomeSiteDoJoin($homeSite) {
+
+        // Allow homesite to be unset. No joining is done since 5.4.
+        // In production $homesite should never be a Site instance, but
+        //  legacy test datasets still invoke this function.
```

Comment:
> legacy test datasets still invoke this function

By legacy test datasets, do you mean those in GOCDB's codebase?

If so, could we remove them? So that this function (or a new function and get rid of this one) only has to handle the `null` case of deleting the legacy association?

Reply:
For legacy read https://github.com/GOCDB/gocdb/blob/7156c2edb671f4023d2bb5f995332aac00b5a7f8/lib/Doctrine/deploy/sampleData/UsersAndRoles.xml#L19 and https://github.com/GOCDB/gocdb/blob/7156c2edb671f4023d2bb5f995332aac00b5a7f8/lib/Doctrine/deploy/AddUsers.php#L51
Everything is possible ;-), but I thought the plan was to wait until there are no users left with homesites before a more thorough clean up?